frame/balances/src/lib.rs

```diff
@@ -1146,9 +1147,10 @@ impl<T: Config<I>, I: Instance> ReservableCurrency<T::AccountId> for Module<T, I
 
 	/// Unreserve some funds, returning any amount that was unable to be unreserved.
 	///
-	/// Is a no-op if the value to be unreserved is zero.
+	/// Is a no-op if the value to be unreserved is zero or the account does not exist.
 	fn unreserve(who: &T::AccountId, value: Self::Balance) -> Self::Balance {
 		if value.is_zero() { return Zero::zero() }
+		if Self::is_dead_account(&who) { return value }
```

Comment:
Why don't we do this check in reserve as well?

Reply:
@pepyakin reserve returns a DispatchResult and will already be a noop if there are not enough funds to reserve. If the value is zero it is also a noop.

Reply:
The functions to worry about were the ones that use `mutate_account` whereas reserve uses `try_mutate_account`... but maybe you see something I dont?

Reply:
nah, that was a genuine question